doc/development-process.md

```diff
@@ -0,0 +1,160 @@
+
+Step one, you **must** have the following things installed:
+
+* [go](https://go.dev) v1.17 -- Primary development language.
+* [docker](https://www.docker.com/) -- Used for generating containers & testing kubernetes set-ups.
+* [helm](https://helm.sh/docs/intro/install/) v3. -- Package manager for kubernetes
+* [kind](https://kind.sigs.k8s.io/docs/user/quick-start#installation) -- Run kubernetes clusters in docker for testing.
+* [flux](https://fluxcd.io/docs/get-started/) -- Continuous delivery system for kubernetes that weave-gitops enriches.
+* [tilt](https://tilt.dev/) -- Automatically build and deploy to a local cluster.
+
+Now you have those things installed, lets go!
+
+## Quick start development setup
+
+Help me get going as fast as possible!
+
+Create a kind cluster. We'll do this so we can have a local cluster on our machine
+
+`$ kind create cluster --name=<some name>`
+
+Then we install flux on it.
+
+`$ flux install`
+
+Then we bring up tilt.
+
+`$ tilt up`
+
+Tilt will tell you the url, or hit the space bar and it'll automagically open a browser for you which will show your local cluster and the resources running. You should see 3 things, Tiltfile, dev-weave-gitops, uncategorised.
+
+Goto login with username: dev, password:dev at http://localhost:9001/sign_in
+
+When you make changes in the code with this setup, you have to wait for k8s to redeploy the updated pod, it isn't lightening instant fast but **it works** to get you going.
+
+Woop! It's working. It'll be empty cos you haven't created any flux objects. You can jump to [
+quickly add a sample workload to UI](#quickly-add-a-sample-workload-to-ui).
+
+
+## Frontend optimised development setup
+
+Help me I just want super fast frontend development!
+
+Step one, make sure you've installed the tools listed at the very top of this file.
+
+Step two, now you **must** **ALSO** have the following installed.
+
+* [node](https://nodejs.org/en/) v16.13.2 - Install Node.js
+
+Now you have those things installed, lets go!
+
+## Quick start frontend development setup
+
+Create a kind cluster. We'll do this so we can have a local cluster on our machine
+
+`$ kind create cluster --name=<some name>`
+
+Then we install flux on it.
+
+`$ flux install`
+
+Then we bring up tilt.
+
+`$ tilt up`
+
+Then we make our node_modules, this will take a little while.
+
+`$ make node_modules`
+
+Then we fire up our frontend server.
+
+`$ npm start`
+
+Goto login with username: dev, password:dev at http://localhost:4567/sign_in
+
+Running this setup is what enables JavaScript hot reloading, it does websocket pushing of your code, so you don't even need to refresh!
+
+Woop! It's working. It'll be empty cos you haven't created any flux objects. You can jump to [quickly add a sample workload to UI](#quickly-add-a-sample-workload-to-ui).
+
+
+## Quickly add a sample workload to UI
+
+To help see some objects in the UI lets create some sample sources, run the following:
+
+`$ flux create source git podinfo \
```

Comment:
<img width="1034" alt="Screenshot 2022-05-11 at 10 07 11" src="https://user-images.githubusercontent.com/19860021/167813266-d47a70bc-19fd-48fb-9f3b-a3a23c333bd9.png">

new lines don't appear to be rendering properly. Could also do:
```bash
<cmd>
``` 
for nice copypasta?

Reply:
ah, I had missed the rendered output. Thanks. I've fixed 👍🏻